From ab17c89078dc225df0255b786dcc55ec39d7ef4f Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Fri, 17 Jan 2020 16:04:06 +0100
Subject: [PATCH] PONRTSYS-6359: Add intel,prx300-bootcore to detection

Make it use the same compatible string as defined in device tree. The
device tree used lantiq,xrx500 before, so this is probably ignored
anyway.
---
 arch/mips/include/asm/mach-lantiq/grx500/lantiq_soc.h | 1 +
 arch/mips/lantiq/grx500_bootcore/prom.c               | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/mips/include/asm/mach-lantiq/grx500/lantiq_soc.h b/arch/mips/include/asm/mach-lantiq/grx500/lantiq_soc.h
index b9101e3412ab..b47873bbf6d2 100755
--- a/arch/mips/include/asm/mach-lantiq/grx500/lantiq_soc.h
+++ b/arch/mips/include/asm/mach-lantiq/grx500/lantiq_soc.h
@@ -42,6 +42,7 @@
 /* SoC compatible */
 #define COMP_GRX500		"lantiq,grx500"
 #define COMP_PRX300		"intel,prx300"
+#define COMP_PRX300_BOOTCORE	"intel,prx300-bootcore"
 
 static inline int ltq_is_grx500(void)
 {
diff --git a/arch/mips/lantiq/grx500_bootcore/prom.c b/arch/mips/lantiq/grx500_bootcore/prom.c
index 112d407d1048..848478e34554 100644
--- a/arch/mips/lantiq/grx500_bootcore/prom.c
+++ b/arch/mips/lantiq/grx500_bootcore/prom.c
@@ -176,7 +176,7 @@ void __init ltq_soc_detect(struct ltq_soc_info *i)
         case SOC_ID_FMX_DPU:
                 i->name = SOC_PRX300;
                 i->type = SOC_TYPE_PRX300;
-                i->compatible = COMP_PRX300;
+                i->compatible = COMP_PRX300_BOOTCORE;
                 break;
 
 #ifdef CONFIG_USE_EMULATOR
